src/libraries/System.Private.CoreLib/src/System/Resources/RuntimeResourceSet.cs

```diff
@@ -309,158 +269,104 @@ private IDictionaryEnumerator GetEnumeratorHelper()
         {
             if (key == null)
                 throw new ArgumentNullException(nameof(key));
-            if (Reader == null || _resCache == null)
+
+            ResourceReader? reader = _defaultReader;
+            Dictionary<string, ResourceLocator>? cache = _resCache;
+            Dictionary<string, ResourceLocator>? caseInsensitiveTable = _caseInsensitiveTable;
+            if (reader is null || cache is null)
                 throw new ObjectDisposedException(null, SR.ObjectDisposed_ResourceSet);
```

Comment:
nit: you may move this condition before the previous  line.

Reply:
That would make the code racy